### PR TITLE
[lldb] Remove Listener::SetShadow

### PR DESCRIPTION
**Pull request by labath**
It's not used since https://reviews.llvm.org/D157556.

### CODE REVIEW

File: lldb/source/Utility/Listener.cpp [hunk 1]
```diff
@@ -18,13 +18,10 @@
...
-Listener::Listener(const char *name)
-    : m_name(name), m_broadcasters(), m_broadcasters_mutex(), m_events(),
-      m_events_mutex(), m_is_shadow() {
+Listener::Listener(const char *name) : m_name(name) {
   Log *log = GetLog(LLDBLog::Object);
-  if (log != nullptr)
-    LLDB_LOGF(log, "%p Listener::Listener('%s')", static_cast<void *>(this),
-              m_name.c_str());
+  LLDB_LOGF(log, "%p Listener::Listener('%s')", static_cast<void *>(this),
+            m_name.c_str());
```
Comment (JDevlieghere) [inline lldb/source/Utility/Listener.cpp:24]: Nit: why not get rid of `log` altogether: 

```
LLDB_LOGF(GetLog(LLDBLog::Object), "%p Listener::Listener('%s')", static_cast<void *>(this),
             m_name.c_str());
```

**Review by JDevlieghere — APPROVED** [review]

**Review by medismailben — APPROVED** [review]
LGTM :)

**Comment by jimingham** [comment]
I have a bug caused by the fact that we aren't properly marking shadow listeners as shadow listeners, causing them to steal events from the primary listener.  I just got it, and haven't started looking at it.  I might need SetShadow for that.

**Comment by jimingham** [comment]
The problem was is currently you cna make a process, then do:

    listener = lldb.SBListener("my_listener")
    process.broadcaster.AddListener(listener, lldb.SBProcess.eBroadcastBitStateChanged)

and nothing marks this as a Shadow listener, so when we go to deliver events, we don't wait on the consumption by the Primary listener.

But I don't think I should need SetShadow for this.  I don't think we need to mark the listener outside the "Add" action as Shadow.  Instead, AddListener should see if there IS a primary listener, and if there is one, then any other listener has to be a shadow listener.

Actually, I wonder if we really need this distinction to be marked explicitly, I might have been being too general.  If you have a primary listener, then all the other listeners have to be shadow listeners.  I don't think anything else makes sense.  And conversely, if you don't have a primary listener, then I marking some of the listeners as shadow doesn't do anything useful.

**Review by jimingham — APPROVED** [review]
Just to be clear, I have no problem with removing this.  Shadow wasn't really a useful property "broadcaster having a primary listener" was the only thing that really mattered.

**Pull request merged**